whois-api/src/main/java/net/ripe/db/whois/api/rest/search/RpkiRoaMessageGenerator.java

```diff
@@ -0,0 +1,59 @@
+package net.ripe.db.whois.api.rest.search;
+
+import com.google.common.collect.ImmutableList;
+import net.ripe.db.whois.api.rest.domain.RpslMessage;
+import net.ripe.db.whois.api.rest.domain.Parameters;
+import net.ripe.db.whois.common.rpki.Roa;
+import net.ripe.db.whois.common.rpki.RpkiDataProvider;
+import net.ripe.db.whois.common.rpki.RpkiService;
+import net.ripe.db.whois.common.rpki.WhoisRoaChecker;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.query.QueryMessages;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+
+@Component
+public class RpkiRoaMessageGenerator implements RpslMessageGenerator {
+
+    private static final ImmutableList<ObjectType> TYPES = ImmutableList.of(ObjectType.ROUTE, ObjectType.ROUTE6);
+    private final boolean isEnabled;
```

Comment:
Class RpkiRoaChecker : Map<Roa, ValidationStatus> validationResult = Maps.newHashMap(); scope should be class level rather than method